components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,933 @@
+    integer :: ierr
+    character(len=150) :: errstr
+    logical :: found
+ 
+    call get_pio_atm_File(trim(pio_atm_filename),pio_atm_file,found)
+    if (.not.found) call errorHandle("PIO Error: Could not register variable"//trim(shortname)//", pio file "//trim(pio_atm_filename)//" not found",-999)
+    ! Get a new variable pointer in var_list
+    curr => pio_atm_file%coord_list_top
+    do while (associated(curr))
+      prev => curr
+      curr => prev%next
+    end do
+    allocate(prev%next)
+    curr => prev%next
+    allocate(curr%coord)
+    hist_coord => curr%coord
+    pio_atm_file%dimcounter = pio_atm_file%dimcounter + 1
+    ! Register this dimension
+    hist_coord%name      = trim(shortname)
+    hist_coord%long_name = trim(longname)
+    hist_coord%dimsize   = length
+    if (length.eq.0) then
+      ierr = PIO_def_dim(pio_atm_file%pioFileDesc, trim(shortname), pio_unlimited , hist_coord%dimid)
+    else
+      ierr = PIO_def_dim(pio_atm_file%pioFileDesc, trim(shortname), length , hist_coord%dimid)
+    end if
+    call errorHandle("PIO ERROR: could not define dimension "//trim(shortname)//" on file: "//trim(pio_atm_filename),ierr)
+    
+    return
+  end subroutine register_dimension
+!=====================================================================!
+  subroutine register_variable(pio_atm_filename,shortname,longname,length,dims,dtype,tag)
+    character(len=*), intent(in) :: pio_atm_filename
+    character(len=*), intent(in) :: shortname,longname
+    integer, intent(in)          :: length
+    character(len=*), intent(in) :: dims(length)
+    integer, intent(in)          :: dtype
+    character(len=*), intent(in) :: tag     ! Unique tag for this variables decomposition type
+
+    ! Local variables
+    type(pio_atm_output),pointer :: pio_atm_file
+    integer :: loc_len
+    logical                      :: found
+    type(io_desc_t), pointer     :: iodesc_loc
+    type(hist_var_t), pointer    :: hist_var
+    integer :: dim_ii
+    integer :: ierr
+    integer, allocatable :: dimlen(:)
+    integer :: total_dimlen, my_dof_len, extra_procs
+    integer :: ii, istart, istop
+    logical :: has_t_dim  ! Logical to flag whether this variable has a time-dimension.  This is important for the decomposition step.
+
+    type(hist_var_list), pointer :: curr => null(), prev => null()
+ 
+    call get_pio_atm_File(trim(pio_atm_filename),pio_atm_file) 
+
+    has_t_dim = .false.
+    pio_atm_file%varcounter = pio_atm_file%varcounter + 1
+
+    ! Get a new variable pointer in var_list
+    curr => pio_atm_file%var_list_top
+    do while (associated(curr))
+      prev => curr
+      curr => prev%next
+    end do
+    allocate(prev%next)
+    curr => prev%next
+    allocate(curr%var)
+    hist_var => curr%var
+    ! Populate meta-data associated with this variable
+    hist_var%name      = trim(shortname)
+    hist_var%long_name = trim(longname)
+    hist_var%numdims   = length
```

Comment:
Another hint that 'length' is a name a bit too generic. "numdims" is already much better.

Reply:
yes, good point.  I'll make the change.